R/model-tree.R

```diff
@@ -381,14 +381,25 @@ make_model_network <- function(full_log){
 
   # Create Network
   start <- "Start"
-  noref <- data.frame(from = NA, to = start)
-  parent_mods <- data.frame(from = start,
-                            to = full_log$run[full_log$based_on==""])
-  child_mods <- data.frame(from = full_log$based_on[full_log$based_on!=""],
-                           to = full_log$run[full_log$based_on!=""])
-  network_df <- rbind(noref, parent_mods, child_mods)
-
-  # Check network links - set unlinked models as starting points
+  noref_df <- data.frame(from = NA, to = start)
+
+  child_mods_df <- data.frame(
+    from = full_log$based_on[full_log$based_on!=""],
+    to = full_log$run[full_log$based_on!=""]
+  )
+
+  parent_mods <- full_log$run[full_log$based_on==""]
+  if(!rlang::is_empty(parent_mods)){
+    parent_mods_df <- data.frame(from = start, to = parent_mods)
+    network_df <- rbind(noref_df, parent_mods_df, child_mods_df)
+  }else{
+    # This happens when there is no "starting" or origin model
+    # - i.e. the first referenced based_on model no longer exists
+    network_df <- rbind(noref_df, child_mods_df)
+  }
+
```

Comment:
Stepping through this for both the origin and no origin cases, this logic made sense to me.